lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -1721,6 +1721,12 @@ D3D_NAME SemanticToSystemValueType(const Semantic *S, DXIL::TessellatorDomain do
     default:
     return D3D_NAME_UNDEFINED;
     }
+  case Semantic::Kind::ShadingRate:
+#define D3D_NAME_CULL_SHADINGRATE 24
+    return (D3D_NAME)D3D_NAME_SHADINGRATE;
+  case Semantic::Kind::CullPrimitive:
+#define D3D_NAME_CULL_PRIMITIVE 25
+    return (D3D_NAME)D3D_NAME_CULL_PRIMITIVE;
```

Comment:
`DxilProgramSigSemantic` is the local equivalent of the `D3D_NAME enum values for this codebase.  We should use those internally for anything relatively newly defined so the reflection implementation doesn't take a dependency on the latest SDK for these.  Use the enums from there for both of these instead of defining values locally, since defining values locally would break if compiled with a newer SDK where they were added to the D3D_NAME enum.

Reply:
It wouldn't break because the local macro would "shadow" the enum value exposed by the header - as long as the `#define` happens *after* the `#include`. Your solution is better though.